main.cpp

```diff
@@ -18,13 +18,24 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private: Rewriter &rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter_):rewriter(rewriter_) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+        const auto *StyleCastExpr=Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        auto leftParenLoc=StyleCastExpr->getLParenLoc();
+        auto rigthParenLoc=StyleCastExpr->getRParenLoc();

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()`.

Reply:
getRParenLos() a special method of the CStyleCastExprclass, this method returns the position where the closing right bracket is located. An object of the CStyleCastExpr class is an indication of a node in the AST tree. 
If call the getSubExprAsWritten() method, it will return the desired tree node, ignoring the auxiliary ImplicitCastExpr nodes.
getBeginLoc() is used here to get the start position of this variable. But need to separately call the method of removing the closing right bracket rewriter_.Remove Text(styleCastExpr->getRParenLoc(), 1); .
The main difference is that when using getRParenLoc(), the search for the desired position is carried out using a closing parenthesis, and when using getSubExprAsWritten(), it is carried out using a variable for which the CStyleCast transformation was called.